packages/drivers/odsp-driver/src/zipItDataRepresentationUtils.ts

```diff
@@ -336,3 +336,64 @@ export class TreeBuilder extends NodeCore {
         return builder;
     }
 }
+
+export function assertBlobCoreInstance(
+    node: NodeTypes,
+    message?: string,
+): asserts node is BlobCore {
+    if (node instanceof BlobCore) {
+        return;
+    }
+    throwBufferParseException(node, "BlobCore", message);
+}
+
+export function assertNodeCoreInstance(
+    node: NodeTypes,
+    message?: string,
+): asserts node is NodeCore {
+    if (node instanceof NodeCore) {
+        return;
+    }
+    throwBufferParseException(node, "NodeCore", message);
+}
+
+export function assertNumberInstance(
+    node: NodeTypes,
+    message?: string,
+): asserts node is number {
+    if (typeof node === "number") {
+        return;
+    }
+    throwBufferParseException(node, "Number", message);
+}
+
+function throwBufferParseException(
+    node: NodeTypes,
+    expectedNodeType: NodeType,
+    message?: string,
+): never {
+    const error = createOdspNetworkError(
+        `BufferParsingException: ${message}`,
+        fetchIncorrectResponse,
+        undefined,
+        undefined,
+        undefined,
+        {
+            nodeType: getNodeType(node),
+            expectedNodeType,
```

Comment:
Is there any way to get a path in the tree logged? It's should be marked as potentially containing PII, but we can solve that problem separately - for DF / 1st party solutions that might be invaluable information to log.

That said, we can solve it separately, even having first level of data is better than nothing.

Reply:
Created an issue here: https://github.com/microsoft/FluidFramework/issues/7186